### DISCUSSION TITLE
PlanPlugin support for Multi-layer Vector Basemaps

### DISCUSSION THREAD

**Discussion by juanbrk** [mistic100/Photo-Sphere-Viewer#1391] [General]
Is it possible to add Multi-layer Vector Basemaps to the PlanPlugin in order to Add the Community Map of Canada Basemap as a layer to the list of available base layers?

The Canadian Topographic basemap is comprised of 4 different layers that make use of leaflet `map panes` in order to group layers together and stack them in a certain order, that are then grouped together using `L.layerGroup` so that Leaflet can interpret them as a singular basemap rather than as separate layers.

 This is a common design for basemaps where multiple sources of topographic information are combined into a unified basemap.

- [Here's a tutorial of what exactly I want to achieve](https://resources.esri.ca/getting-technical/how-to-add-the-community-map-of-canada-basemap-in-leaflet).
- [This is the end result of the map with the multi-layer vector basemap working](https://codepen.io/mwarren_esrica/pen/XWGxWvP)  and the basemap added to Leaflet Layer Controls

In case that's not possible, can I add a single layer from this multi-layer vector basemap as a layer to my list of available base layers? For example this one: [The Canada topographic map](https://www.arcgis.com/home/item.html?id=4dd425da08fe4df7afc30112d6e716be).  I have tried to add It as single layer but failed everytime.

**Comment by mistic100** [comment]
In `configureLeaftlet` you can configure Leaflet freely

https://photo-sphere-viewer.js.org/plugins/plan.html#configureleaflet

**Reply by juanbrk** [reply]
Is there an example on how to actually do this? I'm configuring my map using `configureLeaflet` and end up with two instances of the leaflet map on the page, and can't interact with the `PlanPlugin` one to add any markers or `GeoJson` features to It.

**Reply by mistic100** [reply]
No there is not, if you need further help you have to share your code, and create a new issue.

If you have two leaflet instance it is probably because you create your own, but if you check the doc you will see that `configureLeaflet` already provides an existing instance, and you should interact with this one.

**Reply by mistic100** [reply]
https://github.com/mistic100/Photo-Sphere-Viewer/blob/2d8b0eb2541a529d15d8c57170d476d1153ebbff/examples/plugin-plan.html#L77-L82
